File: 88-merge-sorted-array/merge-sorted-array.js
// Merge Sorted Arrays
// Time: O(n + m)
// Space: O(1)
var merge = function(nums1, m, nums2, n) {
    let idx = nums1.length-1;
    m-=1; n-=1;
    while (m >= 0 || n >= 0) {
        if (m < 0 || (n >= 0 && nums2[n] >= nums1[m])) {
            [nums1[idx], nums2[n]] = [nums2[n], nums1[idx]];
            n--;
        } else if (n < 0 || (m >= 0 && nums1[m] >= nums2[n])) {
            [nums1[idx], nums1[m]] = [nums1[m], nums1[idx]];
            m--;
        }
        idx--;
    }
    return nums1
};

File: 88-merge-sorted-array/merge-sorted-array.swift
class Solution {
    func merge(_ nums1: inout [Int], _ m: Int, _ nums2: [Int], _ n: Int) {
        var i = m-1; 
        var j: Int = n-1
        var idx = nums1.count - 1

        while i >= 0 || j >= 0 {
            if j < 0 {
                nums1[idx] = nums1[i];
                i -= 1;
            } else if i < 0 {
                nums1[idx] = nums2[j];
                j -= 1;
            } else if nums1[i] >= nums2[j] {
                nums1[idx] = nums1[i];
                i -= 1;
            } else if nums2[j] >= nums1[i] {
                nums1[idx] = nums2[j];
                j -= 1;
            }
            idx -= 1;
        }
    }
}